packages/core/src/lib/lib/interactivity.ts

```diff
@@ -68,27 +75,23 @@ export const useEventRaycast = (
   onDestroy(unsubscribePointer)
 
   const onEvent = (e: MouseEvent | PointerEvent) => {
+    const eventType = e.type as keyof ThreltePointerEventMap
     e.preventDefault()
     ctx.pointerOverCanvas.set(true)
     renderCtx.pointerInvalidated = true
     setPointerFromEvent(ctx, e)
 
-    if (rootCtx.interactiveObjects.size === 0 || rootCtx.raycastableObjects.size === 0) return
-    const intersects = runRaycaster(
-      rootCtx,
-      pointer,
-      camera,
-      Array.from(rootCtx.raycastableObjects)
-    )
-    if (intersects.length > 0 && rootCtx.interactiveObjects.has(intersects[0].object)) {
-      getThrelteUserData(intersects[0].object).eventDispatcher?.(
-        e.type as keyof ThreltePointerEventMap,
-        {
-          ...intersects[0],
-          event: e
-        }
-      )
+    const closestIntersection = getClosestIntersection(rootCtx, pointer, camera)
+    if (eventType === 'pointerdown' && e.button === MOUSE.LEFT) {
+      rootCtx.lastPressIntersection = closestIntersection
```

Comment:
- I recently converted this system to hook-based, so this doesn't need to be in the context actually. A less poluted context is better I guess ;)
- I would rename it to something like `pointerDownOn` and just save the object on `pointerdown` event and compare that on `click` event
- `pointerDownOn` might need a reset on `pointerup`, `click` and other events, but I'm not sure of that. (Later me says it doesn't, am I missing something?)

Reply:
I have to admit that I'm not sure what you mean by hook-based in the Svelte context - is the implication that the function's lifecycle is tied to the lifecycle of the Canvas component?

Moving the state from the context sounds great - wasn't too happy about putting it there in the first place. Just to clarify, you're suggesting putting it in the top of the `useEventRaycast` scope instead, along `camera` and `pointer`?

It doesn't need a reset for the sake of this feature - but with a name like `pointerDownOn`, it probably should be reset as the name would be a bit misleading otherwise. I'll take care of it.

Reply:
Yep, exactly. just a regular `let lastPointerDownOn: Object3D | undefind = undefined` in `useEventRaycast` is perfectly fine 👍

Reply:
Probably best to store the instanceId also, so that we can compare that like you do in `targetChanged`?

Reply:
Good one, yes!

Reply:
Fixed 👍